build.gradle

```diff
@@ -215,17 +215,117 @@ def getProjectVersion() {
+        testCompile group: 'io.pravega', name: 'pravega-test-testcommon', version: pravegaVersion
+    }
+
+    javadoc {
+        title = "Serializers avro"
+        dependsOn delombok
+        source = delombok.outputDir
+        failOnError = true
+        exclude "**/impl/**";
+        options.addBooleanOption("Xdoclint:all,-reference", true)
+    }
+
+    jar {
+        from {
+            configurations.compile.collect { it.isDirectory() ? it : zipTree(it) }
+        }
+    }
+}
+
+project('serializers:protobuf') {
+    dependencies {
+        compile project(':serializers:shared')
         compile group: 'com.google.protobuf', name: 'protobuf-java', version: protobufProtocVersion
         compile group: 'com.google.protobuf', name: 'protobuf-java-util', version: protobufUtilVersion
-        compile group: 'io.pravega', name: 'pravega-client', version: pravegaVersion
-        compile group: 'org.xerial.snappy', name: 'snappy-java', version: snappyVersion
-        compile group: 'com.fasterxml.jackson.module', name: 'jackson-module-jsonSchema', version: jacksonVersion
+        testCompile project(':serializers:shared')
+        testCompile group: 'org.slf4j', name: 'log4j-over-slf4j', version: slf4jApiVersion
+        testCompile group: 'ch.qos.logback', name: 'logback-classic', version: qosLogbackVersion
+        testCompile group: 'io.pravega', name: 'pravega-test-testcommon', version: pravegaVersion
+    }
+
+    javadoc {
+        title = "Serializers protobuf"
+        dependsOn delombok
+        source = delombok.outputDir
+        failOnError = true
+        exclude "**/impl/**";
+        options.addBooleanOption("Xdoclint:all,-reference", true)
+    }
+
+    jar {
+        from {
+            configurations.compile.collect { it.isDirectory() ? it : zipTree(it) }
+        }
+    }
+}
+
+project('serializers:json') {
+    dependencies {
+        compile project(':serializers:shared')
         compile group: 'com.github.everit-org.json-schema', name: 'org.everit.json.schema', version: everitVersion
+        compile group: 'com.fasterxml.jackson.module', name: 'jackson-module-jsonSchema', version: jacksonVersion
+        testCompile project(':serializers:shared')
+        testCompile group: 'org.slf4j', name: 'log4j-over-slf4j', version: slf4jApiVersion
+        testCompile group: 'ch.qos.logback', name: 'logback-classic', version: qosLogbackVersion
+        testCompile group: 'io.pravega', name: 'pravega-test-testcommon', version: pravegaVersion
+    }
+
+    javadoc {
+        title = "Serializers json"
+        dependsOn delombok
+        source = delombok.outputDir
+        failOnError = true
+        exclude "**/impl/**";
+        options.addBooleanOption("Xdoclint:all,-reference", true)
+    }
+
+    jar {
+        from {
+            configurations.compile.collect { it.isDirectory() ? it : zipTree(it) }
+        }
+    }
+}
+
+project('serializers') {
```

Comment:
Also, from the description and a few package names I checked, you are not adding `json`, `avro`, or `protobuf` to the package names, is it right? Is there any downside of doing it this way? I can't think of anything right now, but I wonder if I'm missing it.

Reply:
yes, the first comment is correct, users intending to use a specific format only need to depend on the library for that format. 

as for package naming, i deliberately did not include the names avro, protobuf etc in the package name so that we can leverage "split packages" from java. This allows us to keep several classes as package private and not necessarily make them public.

However, the downside is, split package is prohibited from java 9. 
So we may be forced to avoid split packages.

Reply:
On second thought, in future we may choose to move to higher versions of java that do not support `split package`. so i am changing the code to include module name in the package to differentiate.
the only trouble is all common classes that were earlier package private will now have to be public.

Reply:
I built the project and found the following serializer libraries, FYI: 
```
pravega-serializers-xxx.jar
pravega-serializers-shared-xxx.jar
pravega-serializers-avro-xxx.jar
pravega-serializers-protobuf-xxx.jar
pravega-serializers-json-xxx.jar
```